lib/capybara/node/finders.rb

```diff
@@ -166,7 +166,13 @@ def all(*args)
       # @return [Capybara::Node::Element]            The found element or nil
       #
       def first(*args)
+        if Capybara.first_default_waiting
+          options = if args.last.is_a?(Hash) then args.pop.dup else {} end
+          args.push({minimum: 1}.merge(options))
+        end
         all(*args).first
+      rescue Capybara::ExpectationNotMet
+        nil
```

Comment:
Am I correct in assuming that this is for backwards compatibility? Or is there some other reason why this method can not raise the `Capybara::ExpectationNotMet`?

I'm afraid that returning `nil` with `wait_on_first_by_default` could still cause confusion.